app/models/sidebar_url.rb

```diff
@@ -8,13 +8,21 @@ class SidebarUrl < ActiveRecord::Base
   validate :path_validator
 
   def path_validator
-    Rails.application.routes.recognize_path(value)
+    if external?
+      raise ActionController::RoutingError if value !~ Discourse::Utils::URI_REGEXP
+    else
+      Rails.application.routes.recognize_path(value)
+    end
   rescue ActionController::RoutingError
     errors.add(
       :value,
       I18n.t("activerecord.errors.models.sidebar_section_link.attributes.linkable_type.invalid"),
     )
   end
+
+  def external?
+    value.start_with?("http://", "https://")
```

Comment:
Same concern as https://github.com/discourse/discourse/pull/20503/files#r1123930444. I think whether a link is considered external or not should be based on whether the hostname matches and not based on the URL's protocol.

Reply:
Since this is used as part of the validator, I believe we can't just rely on the protocol here.